### ISSUE TITLE
`ccp_write_io` from `avr/cpufunc.h` missing volatile

### ISSUE THREAD

**Issue by sprintersb** [avrdudes/avr-libc#906]
`ccp_write_io` from `avr/cpufunc.h` has the following prototype:
```c++
void ccp_write_io (uint8_t *__ioaddr, uint8_t __value);
```
but it should be
```c++
void ccp_write_io (uint8_t volatile *__ioaddr, uint8_t __value);
```
So that code that passes a SFR will compile without warning.

```c++
#include <avr/cpufunc.h>
#include <avr/io.h>

int main()
{
    ccp_write_io (&PORTB.OUT, 0);
}
```

```
> avr-gcc foo.c -fsyntax-only -mmcu=atxmega128a3u
```

```
foo.c: In function 'main':
foo.c:6:19: warning: passing argument 1 of 'ccp_write_io' discards 'volatile' qualifier from pointer target type [-Wdiscarded-qualifiers]
     ccp_write_io (&PORTB.OUT, 0);
In file included from foo.c:1:
.../avr/include/avr/cpufunc.h:91:29: note: expected 'uint8_t *' {aka 'unsigned char *'} but argument is of type 'volatile register8_t *' {aka 'volatile unsigned char *'}
 void ccp_write_io (uint8_t *__ioaddr, uint8_t __value);
                    ~~~~~~~~~^~~~~~~~
```

Also I am not sure whether this function is outdated, as there is also `_PROTECTED_WRITE` in `avr/mega.h` (included by `avr/io.h`).

So it would be a good idea if the documentatio could sort that out.

**Comment by sprintersb** [comment]
Also, there is no `extern "C"` wrapper for C++, which means using `ccp_write_io` will lead to an undefied referece error.

**Comment by sprintersb** [comment]
Fixed by f960236b1a229b69c9898b51358760af27b22233.
